### PR TITLE
Fix docker driver cert copy on ubuntu's snap package manager

### LINKED ISSUE #10020: fix minikube installation using gcloud snap
https://cloud.google.com/sdk/docs/downloads-snap

gcloud components install minikube

### PR DESCRIPTION
**Pull request by spowelljr**
Fixes #10020

### CODE REVIEW

File: pkg/minikube/command/kic_runner.go [hunk 1]
```diff
@@ -185,7 +186,11 @@ func (k *kicRunner) Copy(f assets.CopyableFile) error {
 		}
 	}
 	klog.Infof("%s (temp): %s --> %s (%d bytes)", k.ociBin, src, dst, f.GetLength())
-	tf, err := ioutil.TempFile("", "tmpf-memory-asset")
+	tmpFolder := ""
+	if isSnapBinary() {
+		tmpFolder = os.Getenv("HOME")
```
Comment (medyagh) [inline pkg/minikube/command/kic_runner.go:191]: lets use a more safe way of getting the home 
how about using  
```
home, err := os.UserHomeDir()

```

File: pkg/minikube/command/kic_runner.go [hunk 2]
```diff
@@ -185,7 +186,11 @@ func (k *kicRunner) Copy(f assets.CopyableFile) error {
 		}
 	}
 	klog.Infof("%s (temp): %s --> %s (%d bytes)", k.ociBin, src, dst, f.GetLength())
-	tf, err := ioutil.TempFile("", "tmpf-memory-asset")
+	tmpFolder := ""
+	if isSnapBinary() {
```
Comment (medyagh) [inline pkg/minikube/command/kic_runner.go:190]: add comment with the link to the issue, that why we are doing this.

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/command/kic_runner.go (hunk 1 above)
Comment (spowelljr) [reply pkg/minikube/command/kic_runner.go]: Good idea, done

File: pkg/minikube/command/kic_runner.go (hunk 2 above)
Comment (spowelljr) [reply pkg/minikube/command/kic_runner.go:193]: Done

**Review by medyagh — APPROVED** [review]

**Pull request merged**